/* SPDX-License-Identifier: Apache-2.0
 * Copyright(c) 2019 Networked Wireless Systems Lab (NeWS Lab), IIT Hyderabad, India
 */

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.ngcode";
option java_outer_classname = "NGcodeProto";
option objc_class_prefix = "HLW";

package ngcode;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc AmfAusfInteraction(AmfAusfRequest) returns (AmfAusfReply) {}
  rpc AmfSmfInteraction(AmfSmfRequest) returns (AmfSmfReply) {}
  
}

// The request message containing the user's name.
message AmfAusfRequest {
  string name = 1;
  string type = 2;
  int64 imsi = 3;
  int64 plmn_id = 4;
  int64 num_autn_vec = 5;
  int64 nw_type = 6;
  int64 msg_type = 7;
  int64 mmei = 8;
  int32 num = 9;
}

// The response message containing the greetings
message AmfAusfReply {
  string message = 1;
  int64 autn_num = 2;
  int64 rand_num = 3;
  int64 xres = 4;
  int64 k_asme = 5;
  int64 default_apn = 6;
  int64 msg_type = 7;
  int32 count = 8;
}

  message AmfSmfRequest {

  int64 guti=1;
  int64 imsi=2;
  int64 s11_cteid_amf=3;
  int64 eps_bearer_id=4;
  int64 apn_in_use=5;
  int64 tai=6;
	int32 s1_uteid_dl = 7;
	string g_trafmon_ip_addr = 8;
	int64 g_trafmon_port = 9;
	int64 msg_type = 10;

}

message AmfSmfReply {
  int64 xres = 1;
	int64 guti = 2;
	int64 imsi = 3;
	int64 apn_in_use = 4;
	int64 tai = 5;
	int64 k_enodeb = 6;
  int64 k_nas_enc = 7;
	int64 k_nas_int = 8;
	int64 tau_timer = 9;
	int32 s11_cteid_amf = 10;
	int32 s11_cteid_upf = 11;
	int32 s1_uteid_ul = 12;
	int32 nw_capability = 13;
	int32 eps_bearer_id = 14;
	int32 e_rab_id = 15;
	string upf_smf_ip_addr = 16;
	int64 upf_smf_port = 17;
	string ue_ip_addr = 18;
	int64 tai_list_size = 19;
	bool res = 20;
  string g_upf_s1_ip_addr = 21;
  int64 g_upf_s1_port = 22;
	int32 s1_uteid_dl = 23;
	string g_trafmon_ip_addr = 24;
	int64 g_trafmon_port = 25;
	int64 msg_type = 26;
}
